### PR TITLE
feat: added SDR Handbook to the repo

### PR DESCRIPTION
**Pull request by maxcamp**
### What does this PR do?
This PR adds Bryan's SDR Handbook as a markdown document to this repo. As part of the work for this PR, internal-only discussion and information was separated from the document. The internal information can now be found at https://salesforce.quip.com/NEs5AiGgL219. The contents of that file will replace the current SDR Handbook document once this PR is approved. 

### What issues does this PR fix or reference?
@W-9642878@

### Functionality Before
The SDR Handbook was only visible to those with access to the Quip document.

### Functionality After
The SDR Handbook is visible to anyone viewing this repo.

### CODE REVIEW

File: HANDBOOK.md [hunk 1]
```diff
@@ -0,0 +1,962 @@
+# Source-Deploy-Retrieve (SDR) Handbook
+
+*To library maintainers, contributors, and consumers:*
+
+*Bryan Powell wrote this information in June 2021 to provide a guide for both how SDR works and how to use it effectively. This is meant to be a living document and should change as contributions are added to the library. If you notice parts of it become outdated or are inaccurate in some way, please update it or start a discussion if needed. If you are merging a pull request that could either have an impact on information in here, or is worth adding new information to the document, it would be much appreciated by future readers to do so.*
+
+
+## Table of Contents
+
+* [Introduction](https://salesforce.quip.com/lBbbAkn8eBLS#OYPACAxbyX9)
```
Comment (AnanyaJha) [inline HANDBOOK.md:10]: Should these be Quip links?

Comment (maxcamp) [reply HANDBOOK.md]: Realized that Quip does not use markdown formatting for intra-document links! Fixed now

File: HANDBOOK.md [hunk 2]
```diff
@@ -0,0 +1,962 @@
+# Source-Deploy-Retrieve (SDR) Handbook
+
+*To library maintainers, contributors, and consumers:*
+
+*Bryan Powell wrote this information in June 2021 to provide a guide for both how SDR works and how to use it effectively. This is meant to be a living document and should change as contributions are added to the library. If you notice parts of it become outdated or are inaccurate in some way, please update it or start a discussion if needed. If you are merging a pull request that could either have an impact on information in here, or is worth adding new information to the document, it would be much appreciated by future readers to do so.*
```
Comment (jag-sfdc) [inline HANDBOOK.md:5]: I don't think we need this section. This is recommended for internal quip docs and doesn't have to be part of the handbook.

Comment (maxcamp) [reply HANDBOOK.md]: Sounds good, removed!

**Review by WillieRuemmele — APPROVED** [review]

**Pull request merged**